rustyms/src/identification/formats/msfragger.rs

```diff
@@ -0,0 +1,165 @@
+            let head_trim = head.as_str().trim();
+            Ok((
+            if head_trim.eq_ignore_ascii_case("N-term") {
+                SequencePosition::NTerm
+            } else if head_trim.eq_ignore_ascii_case("C-term") {
+                SequencePosition::CTerm
+            } else {
+                // Format: `14M` so take only the numeric part
+                head.as_str()[..head.len()-1].trim().parse::<usize>().map(|i| SequencePosition::Index(i-1)).map_err(|err| CustomError::error(
+                    "Invalid FragPipe modification location",
+                    format!("The location number {}", explain_number_error(&err)),
+                    head.context(),
+                ))?
+            },
+            Modification::sloppy_modification(tail.full_line(), tail.location.clone(), None, custom_database)?
+        ))
+        } else {
+            Err(CustomError::error(
+                "Invalid FragPipe modification",
+                "The format `location(modification)` could not be recognised",
+                m.context(),
+            ))
+        }).collect::<Result<Vec<_>, _>>();
+
+        // modification_info,  best_locs, localization_scores
+    }
+    optional { }
+
+    fn post_process(_source: &CsvLine, mut parsed: Self, _custom_database: Option<&CustomDatabase>) -> Result<Self, CustomError> {
+        for (location, modification) in &parsed.modification_info {
+            parsed.peptide.add_simple_modification(*location, modification.clone());
+        }
+        Ok(parsed)
+    }
+);
+
+impl From<MSFraggerData> for IdentifiedPeptidoform {
+    fn from(value: MSFraggerData) -> Self {
+        Self {
+            score: Some(value.hyperscore / 100.0),
+            local_confidence: None,
+            metadata: MetaData::MSFragger(value),
+        }
+    }
+}
+
+/// All possible MSFragger versions
+#[derive(
+    Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default, Serialize, Deserialize,
+)]
+pub enum MSFraggerVersion {
+    /// Current MSFragger version
+    #[default]
+    V4_2,
+}
+
+impl std::fmt::Display for MSFraggerVersion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(f, "{}", self.name())
+    }
+}
+
+impl IdentifiedPeptidoformVersion<MSFraggerFormat> for MSFraggerVersion {
+    fn format(self) -> MSFraggerFormat {
+        match self {
+            Self::V4_2 => VERSION_V4_2,
+        }
+    }
+    fn name(self) -> &'static str {
+        match self {
+            Self::V4_2 => "v4.2",
+        }
+    }
+}
+
+/// The only supported format for MSFragger data
+pub const VERSION_V4_2: MSFraggerFormat = MSFraggerFormat {
+    version: MSFraggerVersion::V4_2,
+    scan_number: "scannum",
+    mass: "precursor_neutral_mass",
+    rt: "retention_time",
+    z: "charge",
+    ion_mobility: "ion_mobility",
+    rank: "hit_rank",
+    peptide: "peptide",
+    protein: "proteins",
+    theoretical_mass: "calc_neutral_pep_mass",
```

Comment:
I like to not store these things, these can easily be calculated based on the definition, so storing them just takes unnecessary space. (same for `mass` and `delta_mass`, unless `mass` is the experimental mass then it adds stuff that cannot be otherwise calculated, important here though is that I expect the experimental mass in MH+ and it looks like M here)